blog/2024-06-20-help-us-test-opentofu-1-8-0-alpha1.md

```diff
@@ -0,0 +1,109 @@
+
+For the releases above, please unpack the archive and you should find the `tofu` binary inside. You can also use the [standalone installer](https://opentofu.org/docs/intro/install/standalone/) to download the release with signature verification.
+
+## Early variable/locals evaluation
+
+This feature lets you use variables and locals for **backends**, **module sources** and **encryption configuration** as long as they are not dependent on resources, data sources or module outputs. This works even if a local is referencing a variable, for example. This is only the first in a series of improvements that will make the .tf code more flexible with more improvements coming in future releases.
+
+The `tofu init` command will now consume your `.tfvars` file and let you specify variables using the `-var` and `-var-file` options. Please note that this alpha release will *not* prompt you for missing variables, which is a feature we will add later.
+
+For example, if you wanted to use the same configuration for your S3 backend and your AWS provider, you can now do this:
+
+```hcl
+variable "aws_region" {
+  default = "us-east-1"
+}
+
+terraform {
+  backend "s3" {
+    region = var.aws_region
+  }
+}
+
+provider "aws" {
+  region = var.aws_region
+}
+```
+
+You can also use this to manage module versions with both registry references and git URLs.
+
+```hcl
+locals {
+  aws_module_version = "5.6.1"
+}
+
+module "ec2_instance" "webserver" {
+  source  = "terraform-aws-modules/ec2-instance/aws"
+  version = local.aws_module_version
+
+  // Other ec2_instance options
+}
+
+module "ec2_instance" "db" {
+  source  = "https://github.com/terraform-aws-modules/terraform-aws-ec2-instance?ref=v${local.aws_module_version}"
+
+  // Other ec2_instance options
+}
+```
+
+Finally, here's how you can set up encryption with a passphrase using a variable:
+
+```hcl
+variable "passphrase" {
+  type = string
+}
+
+terraform {
+  encryption {
+    key_provider "pbkdf2" "my_passphrase" {
+      passphrase = var.passphrase
+    }
+
+    method "aes_gcm" "my_method" {
+      keys = key_provider.pbkdf2.my_passphrase
+    }
+
+    state {
+      method = method.aes_gcm.my_method
+    }
+  }
+}
+```
+
+## Override files for OpenTofu: keeping compatibility
+
+Since we are now adding features to OpenTofu that are not present in Terraform, we want to give module authors the ability to write code for both OpenTofu and Terraform without needing to maintain two copies of their modules. You can now create files named `.tofu` that are exclusive to OpenTofu. If you create a file named `foo.tofu`, OpenTofu will ignore the similarly-named `foo.tf` file. You can use this functionality to put your Terraform-specific code in the `.tf` file and then override it for OpenTofu in the `.tofu` file.
+
```

Comment:
Should we mention override for tests anywhere? It's not as much experimental, but still part of the release

Reply:
Yes, it was added in the PR (https://github.com/opentofu/opentofu/pull/1682) and here is the actual source file: https://github.com/opentofu/opentofu/blob/main/website/docs/cli/commands/test/index.mdx